### PR TITLE
Improvements/error management

### PR DESCRIPTION
**Pull request by jocgir**
Some errors were not reported correctly. If a system call returned multi-line error, only the first line was kept.
struct keyword has been added as a go reserved keyword (this avoided using a variable named struct)

### CODE REVIEW

File: template/razor_test.go [hunk 1]
```diff
@@ -504,3 +504,59 @@ func TestMultilineStringProtect(t *testing.T) {
...
+	}{
+		{"Empty", `@data("")`, "", nil},
+		{"Integer", `@data("1")`, "1", nil},
+		{"Hcl", `@data("a = 1 b = 2")`, "a=1 b=2", nil},
+		{"Hcl type", `@typeOf(data("a = 1 b = 2"))`, "hcl.hclDict", nil},
+		{"Hcl kind", `@kindOf(data("a = 1 b = 2"))`, "map", nil},
+		{"Invalid", "@typeOf(data(`\"a\": 1, \"b\": 2`))", `"<RUN_ERROR>"`, fmt.Errorf("")},
+		{"Json", "@typeOf(data(`{\"a\": 1, \"b\": 2}`))", "json.jsonDict", nil},
+		{"Yaml", "@typeOf(data(`a: 1\nb: 2`))", "yaml.yamlDict", nil},
+		{"Flexible Hcl", "@typeOf(data(`a = 1 b = hello`))", "yaml.yamlDict", nil}, // TODO: Change that to hcl
```
Comment (julienduchesne) [inline template/razor_test.go]: TODO for now or later?

File: template/razor_repl_expr.go [hunk 2]
```diff
@@ -17,18 +17,25 @@ const (
 	literalTripleBackticks = "_=!TRIPLE_BT!=_"
```
Comment (julienduchesne) [inline template/razor_repl_expr.go]: `@` and triple backticks can't go in the reservedKeywords slice below?

Comment (jocgir) [reply template/razor_repl_expr.go]: I don't think so. Too risky to induce side effects IMO.

File: template/razor_test.go (hunk 1 above)
Comment (jocgir) [reply template/razor_test.go]: Shit, I really don't remember. I merged an old branch that has never been submitted as a PR.
I would say later. I think that I wrote that test when I started to look at HCL2 when it was still in Beta IIRC.

**Pull request merged**